File: problems/383. Ransom Note/Solution.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        std::map<char, int> magazineMap;
        for (const char& ch : magazine) {
            magazineMap[ch] += 1;
        }
        for (const char& ch : ransomNote) {
            if (magazineMap.count(ch) == 0 || magazineMap[ch] == 0) return false;
            magazineMap[ch] -= 1;
        }
        return true;
    }
};

File: problems/383. Ransom Note/Solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        table = {}
        for char in magazine:
            if char not in table: table[char] = 1
            else: table[char] += 1
        for char in ransomNote:
            if (char not in table) or (table[char] == 0): return False
            table[char] -= 1
        return True